src/ImgCrop.tsx

```diff
@@ -314,6 +334,69 @@ const ImgCrop = forwardRef<CropperRef, ImgCropProps>((props, cropperRef) => {
   const title = modalTitle || (isCN ? '编辑图片' : 'Edit image');
   const resetBtnText = resetText || (isCN ? '重置' : 'Reset');
 
+  // Exposes `editFile` method to parent via ref, allowing programmatic cropping
+  const editFile = useCallback(
+    (file: RcFile | UploadFile) => {
+      const reader = new FileReader();
+
+      // Once the file is loaded as DataURL, open the crop modal
+      reader.onload = () => {
+        if (typeof reader.result !== 'string') return;
+
+        // Display modal with the loaded image
+        setModalImage(reader.result);
+
+        // Handler: user cancels cropping
+        onCancel.current = () => {
+          setModalImage('');
+          easyCropRef.current?.onReset();
+        };
+
+        // Handler: user confirms cropping
+        onOk.current = async (event: MouseEvent<HTMLElement>) => {
+          setModalImage('');
+          easyCropRef.current?.onReset();
+
+          // Get cropped canvas from cropper instance
+          const canvas = getCropCanvas(event.target as ShadowRoot);
+          const { type = 'image/jpeg', name = 'cropped.jpg', uid } = file;
+
+          // Convert canvas to blob and wrap it as a File
+          canvas.toBlob(
+            async (croppedBlob) => {
+              if (!croppedBlob) return;
+              const newFile = new File([croppedBlob], name, { type });
+
+              // Preserve UID so Upload list can replace the old file correctly
+              Object.assign(newFile, { uid });
+
+              // Trigger onModalOk callback with cropped file
+              cb.current.onModalOk?.(newFile);
```

Comment:
[Question] wondering why this work because we're not using `onModalOk` prop in our project 🤔
Should we wrap by `runBeforeUpload`?

https://github.com/smartnews/antd-img-crop/blob/9f8631af294f7ecec19af5a2d17ba6edb75e4ab2/src/ImgCrop.tsx#L262-L290

Reply:
I think you're onto something, yes, thanks for noticing!
Let me check. We need a way to trigger some kind of upload after cropping 👌